src/components/StatusFilter.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <el-select v-model="value" filterable placeholder="Filter by status">
+    <el-option 
+      v-for="item in options"
+        :key="item.value"
+        :label="item.label"
+        :value="item.value"
+    ></el-option>
+  </el-select>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+@Component
+export default class StatusFilter extends Vue {
+  @Prop() private options: Array<object> = [
```

Comment:
We know all types of object, better create a custom type `items: Array<FilterOptions>`